tests/bench_runtime_coreapis/main.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Measure runtime of select core API functions
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "mutex.h"
+#include "benchmark.h"
+#include "thread.h"
+#include "thread_flags.h"
+
+#ifndef BENCH_RUNS
+#define BENCH_RUNS          (1000UL * 1000UL)
+#endif
+
+static mutex_t _lock;
+static thread_t *t;
+static thread_flags_t _flag = 0x0001;
+static msg_t _msg;
+
+static void _mutex_lockunlock(void)
+{
+    mutex_lock(&_lock);
+    mutex_unlock(&_lock);
+}
+
+static void _flag_waitany(void)
+{
+    thread_flags_set(t, _flag);
+    thread_flags_wait_any(_flag);
+}
+
+static void _flag_waitall(void)
+{
+    thread_flags_set(t, _flag);
+    thread_flags_wait_all(_flag);
+}
+
+static void _flag_waitone(void)
+{
+    thread_flags_set(t, _flag);
+    thread_flags_wait_one(_flag);
+}
+
+int main(void)
+{
+    puts("Runtime of Selected Core API functions\n");
+
+    t = (thread_t *)sched_active_thread;
+
+    BENCHMARK_FUNC("nop loop", BENCH_RUNS, __asm__ volatile ("nop"));
```

Comment:
A different way to write this which does not require a nop instruction is to use
```c
__asm__ volatile ("" ::: "memory")
```

Reply:
@gebart what are the advantages of this? I prefer `nop` for readability, but does it take some finite amount of time? (this is here to assess how long the stack push/pop and context switch takes, right?)

Reply:
The main benefit is that there is really nothing in the loop, and there is no architecture specific asm code. But I guess most architectures have a nop instruction, so it is probably not significant.